test/data/toBundleExamples/circular_reference_inline/expected.json

```diff
@@ -0,0 +1,82 @@
+{
+  "openapi": "3.0.2",
+  "info": {
+    "version": "1.0.0",
+    "title": "Swagger Petstore",
+    "description": "A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification",
+    "termsOfService": "http://swagger.io/terms/",
+    "contact": {
+      "name": "Swagger API Team",
+      "email": "apiteam@swagger.io",
+      "url": "http://swagger.io"
+    },
+    "license": {
+      "name": "Apache 2.0",
+      "url": "https://www.apache.org/licenses/LICENSE-2.0.html"
+    }
+  },
+  "paths": {
+    "/pets": {
+      "get": {
+        "description": "Returns all pets alesuada ac...",
+        "operationId": "findPets",
+        "responses": {
+          "200": {
+            "description": "An paged array of pets",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/_schemas_schemas.yaml-_components_schemas_ErrorResponse"
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "_schemas_schemas.yaml-_components_schemas_ErrorResponse": {
+        "title": "Error response",
+        "description": "Common error response for all Azure Resource Manager APIs to return error details for failed operations. (This also follows the OData error response format.).",
+        "type": "object",
+        "properties": {
+          "error": {
+            "type": "object",
+            "description": "The error detail.",
+            "properties": {
+              "code": {
+                "readOnly": true,
+                "type": "string",
+                "description": "The error code."
+              },
+              "message": {
+                "readOnly": true,
+                "type": "string",
+                "description": "The error message."
+              },
+              "target": {
+                "readOnly": true,
+                "type": "string",
+                "description": "The error target."
+              },
+              "details": {
+                "readOnly": true,
+                "type": "array",
+                "items": {
+                  "$ref": "/schemas/schemas.yaml#components/schemas/ErrorDetail",
+                  "x-circularRef": true
```

Comment:
@LuisTejedaS We decided that we'll not introduce `x-circularRef` for now and we'll just keep track of all circular refs in memory. i.e. similar to how we have reference map. We can decide to expose it when we decide to but for now just have this info ready to be exposed.

Reply:
Hi  @VShingala  Can you please elaborate more on "similar to how we have reference map" are you thinking in a particular structure? What information is going to be needed?

Reply:
@LuisTejedaS A simple array of all circular references should do it. As bundled data will already have `$ref` values, we can detect from this list of circular refs, which `$ref` are circular.

So for example, how we store `referenceMap`, we can store similar field named `circularRefs` under `contentAndComponents` and expose it similarly if needed.

i.e. structure of `circularRefs`:

```
[
  "#/components/schemas/ErrorDetails",
  ...
]
```

Reply:
Thanks @VShingala I added the array and right now is not published via result